crates/mako/src/visitors/env_replacer.rs

```diff
@@ -4,172 +4,122 @@ use std::sync::Arc;
-                                } else {
-                                    // replace with `undefined` if env not found
-                                    *expr = *Box::new(Expr::Ident(Ident::new(
-                                        js_word!("undefined"),
-                                        DUMMY_SP,
-                                    )));
-                                }
-                            }
+                        MemberProp::Ident(Ident { sym, .. }) => {
+                            member_visit_path.push('.');
+                            member_visit_path.push_str(sym.as_ref());
                         }
+                        MemberProp::Computed(ComputedPropName { expr, .. }) => {
+                            match expr.as_ref() {
+                                Expr::Lit(Lit::Str(Str { value, .. })) => {
+                                    member_visit_path.push('.');
+                                    member_visit_path.push_str(value.as_ref());
+                                }
 
-                        MemberProp::Ident(Ident { sym, .. }) => {
-                            if let Some(env) = EnvReplacer::get_env(&envs, sym) {
-                                // replace with real value if env found
-                                *expr = env;
-                            } else {
-                                // replace with `undefined` if env not found
-                                *expr = *Box::new(Expr::Ident(Ident::new(
-                                    js_word!("undefined"),
-                                    DUMMY_SP,
-                                )));
+                                Expr::Lit(Lit::Num(Number { value, .. })) => {
+                                    member_visit_path.push('.');
+                                    member_visit_path.push_str(&value.to_string());
+                                }
+                                _ => (),
                             }
                         }
                         _ => {}
                     }
+                    current_member_obj = obj.as_ref();
                 }
-            } else if let Expr::Member(MemberExpr {
-                obj:
-                    box Expr::MetaProp(MetaPropExpr {
-                        kind: MetaPropKind::ImportMeta,
-                        ..
-                    }),
+
+                if let Expr::Ident(Ident { sym, .. }) = current_member_obj {
+                    member_visit_path.push('.');
+                    member_visit_path.push_str(sym.as_ref());
+                }
+                let member_visit_path = member_visit_path
+                    .split('.')
+                    .rev()
+                    .collect::<Vec<&str>>()
+                    .join(".");
+
+                if let Some(env) = self.get_define_env(&member_visit_path) {
+                    *expr = env
+                }
+            }
+            Expr::Member(MemberExpr {
```

Comment:
try case 
```
process.env["A"]
process["env"].A
```
i think those two case can be handled in an `Expr::Member(MemberExpr)` match